libbeat/processors/syslog/syslog_test.go

```diff
@@ -28,22 +28,28 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common/cfgtype"
 )
 
-func mustParseTime(layout string, value string) time.Time {
-	t, err := time.Parse(layout, value)
-	if err != nil {
-		panic(err)
+// mustParseTime will parse value into a time.Time using the provided layout. If value
+// cannot be parsed, this function will panic. If layout does not specify a time zone,
+// then a time.Location should be provided by loc. If layout does specify a time zone,
+// then loc should be nil. Layouts that do not specify a year will be enriched with
+// the current year relative to the location specified for the parsed timestamp.
+func mustParseTime(layout, value string, loc *time.Location) time.Time {
+	var t time.Time
+	var err error
+
+	if loc != nil {
+		t, err = time.ParseInLocation(layout, value, loc)
+	} else {
+		t, err = time.Parse(layout, value)
 	}
-
-	return t
-}
-
-func mustParseTimeLoc(layout string, value string, loc *time.Location) time.Time {
-	t, err := time.ParseInLocation(layout, value, loc)
 	if err != nil {
 		panic(err)
 	}
-	if layout == time.Stamp {
-		t = t.AddDate(time.Now().In(loc).Year(), 0, 0)
+
+	// Timestamps that do not include a year will be enriched using the
+	// current year relative to the location specified for the timestamp.
+	if t.Year() == 0 {
+		t = t.AddDate(time.Now().In(t.Location()).Year(), 0, 0)
 	}
```

Comment:
This will never be executed since year 0 does not exist (https://play.golang.com/p/fTw5W2EwC3M).

```suggestion
	if t.Year() == 1 {
		t = t.AddDate(time.Now().In(t.Location()).Year()-1, 0, 0)
	}
```
(justification: https://play.golang.com/p/zzX_jnLA2md)

Reply:
That may be the case with zero time (`time.Time{}`), but that's not the case after parsing a timestamp without a year:
https://go.dev/play/p/ziRlcUF8OT2

I haven't dug into the details of Go's time parsing, but I'd hazard a guess that they chose year 0 in this case for the exact reason that year 0 does not exist. It wouldn't be right to give this timestamp a year if one wasn't originally specified. Or maybe it's a coincidence, not sure.